packages/back-end/src/services/stats.ts

```diff
@@ -122,7 +125,11 @@ result = analyze_metric_df(
   weights=weights,
   type=type,
   inverse=inverse,
-  # engine=StatsEngine.FREQUENTIST
+  engine=${
+    statsEngine === "bayesian"
+      ? "StatsEngine.BAYESIAN"
+      : "StatsEngine.FREQUENTIST"
+  }
```

Comment:
I'm curious... we pass all other data to the python script via the `data = json.loads(` call above rather than inline (e.g. see how we load `type = data['type']` or `data['inverse']`). Is there any particular reason we do it that way? Is it more secure than dropping it inline here?

Reply:
This is a good callout, I overlooked that. I think it's important when interpolating string values into the script since they may contain special characters that need escaping/replacing (e.g. line 92).

The modification in this PR is not really introducing a scalar value but rather toggling between two values already in scope, so we don't face the same issue there.

Normally I would  follow the convention anyway, but since the JSON is purely representing scalar values (i.e. it is stringified then parsed) I'm not sure how I could interpolate a reference to  `StatsEngine.BAYESIAN` or the like into it

Reply:
I made an attempt to go the json route - would be good if you could pull down and test locally if you get a chance

https://github.com/growthbook/growthbook/pull/837/commits/2d3c2abb7765a376b63ce847fadfac3e8d48ae8c

Reply:
I undid the change after sync'ing w/ Jeremy - original way was fine and preferred